kura/org.eclipse.kura.core.comm/src/main/java/org/eclipse/kura/core/comm/CommConnectionFactory.java

```diff
@@ -8,49 +8,26 @@
  *
  * Contributors:
  *     Eurotech
+ *     Red Hat Inc - clean up
  *******************************************************************************/
 package org.eclipse.kura.core.comm;
 
+import static org.eclipse.kura.comm.CommURI.parseString;
+
 import java.io.IOException;
 
 import javax.microedition.io.Connection;
 
-import org.eclipse.kura.comm.CommURI;
-import org.osgi.service.component.ComponentContext;
 import org.osgi.service.io.ConnectionFactory;
 
 public class CommConnectionFactory implements ConnectionFactory {
 
-    @SuppressWarnings("unused")
-    private ComponentContext m_ctx;
-
-    // ----------------------------------------------------------------
-    //
-    // Dependencies
-    //
-    // ----------------------------------------------------------------
-
-    // ----------------------------------------------------------------
-    //
-    // Activation APIs
-    //
-    // ----------------------------------------------------------------
-
-    protected void activate(ComponentContext componentContext) {
-        //
-        // save the bundle context
-        this.m_ctx = componentContext;
-    }
-
-    protected void deactivate(ComponentContext componentContext) {
-        this.m_ctx = null;
-    }
-
     @Override
     public Connection createConnection(String name, int mode, boolean timeouts) throws IOException {
         try {
-            CommURI uri = CommURI.parseString(name);
-            return new CommConnectionImpl(uri, mode, timeouts);
+            return new CommConnectionImpl(parseString(name), mode, timeouts);
+        } catch (IOException e) {
+            throw e; // re-throw
         } catch (Throwable t) {
```

Comment:
The idea is to never implementation specific checked or unchecked exceptions escalate to the higher layers. `IOException` occurs at the communication layer and the business layer do not need to know about such exceptions.

In case of such scenario, we could either - 

1. Convert `IOExcepton` to `KuraException`, if the client of this bundle is expected to recover from such exceptional cause
2. Convert `IOException` to `KuraRuntimeException`, if the client of this bundle `cannot` do anything

This is what Kura follows now to separate lower level exceptions from the higher level.